### PR TITLE
[JSON manual] json inserts

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-23492

Add json_array_append, json_array_insert, json_insert, json_set, json_replace, json_remove to manual.

### CODE REVIEW

File: en/sql/function/json_fn.rst [hunk 1]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
 .. function:: JSON_MERGE (json_doc, json_doc [, json_doc] ...)
 
   **JSON_MERGE** is an alias for **JSON_MERGE_PRESERVE**.
+
+JSON_ARRAY_APPEND
+===================================
+
+.. function:: JSON_ARRAY_APPEND (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_ARRAY_APPEND** function appends the given json_vals at their given json paths inside the json_doc.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
  The **JSON_ARRAY_APPEND** function returns a modified copy of first argument. For each given <json path, json_val> pair, the function appends the value to the json array addressed by the corresponding path.
```

File: en/sql/function/json_fn.rst [hunk 2]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+    ======================
+      {"a":[[1,"1"],2]}
+
+
+JSON_ARRAY_INSERT
+===================================
+
+.. function:: JSON_ARRAY_INSERT (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_ARRAY_INSERT** function inserts the given value in the json arrays at the given json paths.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
  The **JSON_ARRAY_INSERT** function returns a modified copy of first argument. For each given <json path, json_val> pair, the function inserts the value in the json array addressed by the corresponding path.
```

File: en/sql/function/json_fn.rst [hunk 3]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+      json_array_insert('"a"', '$[5]', '1')
+    ======================
+      ["a",null,null,null,null,"1"]
+
+JSON_INSERT
+===================================
+
+.. function:: JSON_INSERT (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_INSERT** function inserts json_val arguments inside the json_doc at the given paths.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: The **JSON_INSERT** function returns a modified copy of first argument. For each given <json path, json_val> pair, the function inserts the value if no other value exists at the corresponding path.

File: en/sql/function/json_fn.rst [hunk 4]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+      json_insert('[0,1,2]', '$[4]', '1')
+    ======================
+      [0,1,2,null,"1"]
+
+JSON_SET
+===================================
+
+.. function:: JSON_SET (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_SET** function inserts or replaces json_val arguments inside the json_doc at the given paths.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
  The **JSON_SET** function returns a modified copy of first argument. For each given <json path, json_val> pair, the function inserts or replaces the value at the corresponding path.
```

File: en/sql/function/json_fn.rst [hunk 5]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+    ======================
+      [0,1,2,null,"1"]
+
+
+JSON_REPLACE
+===================================
+
+.. function:: JSON_REPLACE (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_REPLACE** function replaces json elements inside the json_doc at the given paths.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
 The **JSON_REPLACE** function returns a modified copy of first argument. For each given <json path, json_val> pair, the function replaces the value only if another value is found at the corresponding path.
```

File: en/sql/function/json_fn.rst [hunk 6]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+      [0,1,2,null,"1"]
+
+JSON_SET
+===================================
+
+.. function:: JSON_SET (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_SET** function inserts or replaces json_val arguments inside the json_doc at the given paths.
+
+
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: remove one empty line

File: en/sql/function/json_fn.rst [hunk 7]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+===================================
+
+.. function:: JSON_INSERT (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_INSERT** function inserts json_val arguments inside the json_doc at the given paths.
+  Returns the resulting json after insertion.
+
+  The insertion rules for **JSON_INSERT** are the following:
+
+  For each (json path, json_val) pair, from left to right, if the json path addresses an existing element inside the json_doc, the argument pair is ignored and does not overwrite the existing json element.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: This line may be removed if suggested first paragraph is accepted.

File: en/sql/function/json_fn.rst [hunk 8]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+
+JSON_SET
+===================================
+
+.. function:: JSON_SET (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_SET** function inserts or replaces json_val arguments inside the json_doc at the given paths.
+
+
+  For each (json path, json_val) pair, from left to right, an existing value is searched inside the json_doc at the json path and is replaced with the given json_val.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: This line may be removed if suggested first paragraph is accepted.

File: en/sql/function/json_fn.rst [hunk 9]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+  Otherwise, the json_val is inserted if the json path addresses one of the following json values inside the json_doc:
+
+  - An inexistent object member of an existing json object. A (key, value) pair is added to the json object with the key deduced from the json path and the value being the json_val.
+  - An array index past of an existing json array's end. The array is filled with nulls after the initial end of the array and the json_val is inserted at the specified index.
+
+  The document produced by evaluating one pair becomes the new value against which the next pair is evaluated. 
+
+  Returns the resulting json after insertion.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: This line may be removed if suggested first paragraph is accepted.

File: en/sql/function/json_fn.rst [hunk 10]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+
+
+JSON_REPLACE
+===================================
+
+.. function:: JSON_REPLACE (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_REPLACE** function replaces json elements inside the json_doc at the given paths.
+
+  For each (json path, json_val) pair, from left to right, an existing value is searched inside the json_doc at the json path and is replaced with json_val.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: This line may be removed if suggested first paragraph is accepted.

File: en/sql/function/json_fn.rst [hunk 11]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+  Otherwise, if the json_path does not exist inside the json_doc, the (json path, json_val) pair is ignored and has no effect.
+
+  The document produced by evaluating one pair becomes the new value against which the next pair is evaluated. 
+
+  Returns the resulting json after replacements.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: This line may be removed if suggested first paragraph is accepted.

File: en/sql/function/json_fn.rst [hunk 12]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+      json_replace('[0,1,2]', '$[4]', '1')
+    ======================
+      [0,1,2]
+
+JSON_REMOVE
+===================================
+
+.. function:: JSON_REMOVE (json_doc, json path [, json path] ...)
+
+The **JSON_REMOVE** function removes the json element addressed by the json paths in the json_doc.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
The **JSON_REMOVE** function returns a modified copy of first argument, by removing values from all given paths.
```

File: en/sql/function/json_fn.rst [hunk 13]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+
+The json path arguments are evaluated one by one, from left to right. The result produced by evaluating a json path becomes the value against which the next json path is evaluated.
+
+  Returns the resulting json after removal.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: This line may be removed if suggested first paragraph is accepted.

File: en/sql/function/json_fn.rst [hunk 14]
```diff
@@ -748,3 +748,296 @@ JSON_MERGE
...
+  Returns NULL if any argument is NULL.
+  An error occurs if any argument is invalid or if a path points to the root or if it does not exist inside the json_doc.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
  An error occurs if any argument is invalid or if a path points to the root or if a path does not exist.
```

File: en/sql/function/json_fn.rst [hunk 15]
```diff
@@ -748,3 +748,287 @@ JSON_MERGE
...
+JSON_ARRAY_APPEND
+===================================
+
+.. function:: JSON_ARRAY_APPEND (json_doc, json path, json_val [, json path, json_val] ...)
+
+  The **JSON_ARRAY_APPEND** function returns a modified copy of the first argument. For each given <json path, json_val> pair, the function appends the value to the json array addressed by the corresponding path.
+
+  The (json path, json_val) pairs are evaluated one by one, from left to right. The document produced by evaluating one pair becomes the new value against which the next pair is evaluated.
+
+  If the json path points to an json array inside the json_doc, the json_val is appended at the end of the array.
```
Comment (eseokoh) [inline en/sql/function/json_fn.rst]:
```suggestion
  If the json path points to a json array inside the json_doc, the json_val is appended at the end of the array. 
```

**Pull request merged**